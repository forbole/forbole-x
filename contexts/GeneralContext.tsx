import React from 'react';
import get from 'lodash/get';
import currencies from '../misc/currencies';
import usePersistedState from '../misc/usePersistedState';

type Theme = 'light' | 'dark';
type Currency = typeof currencies[number];
type FavValidators = string[];
type FavAddress = { address: string; crypto: string; moniker: string; note?: string; img?: string };
interface GeneralState {
  currency: Currency;
  currencyRate: number; // per USD rate
  theme: Theme;
  alwaysRequirePassword: boolean;
  setAlwaysRequirePassword?: React.Dispatch<React.SetStateAction<boolean>>;
  hideAmount: boolean;
  setHideAmount?: React.Dispatch<React.SetStateAction<boolean>>;
  favValidators: FavValidators;
  favAddresses: FavAddress[];
  setCurrency?: React.Dispatch<React.SetStateAction<Currency>>;
  setTheme?: React.Dispatch<React.SetStateAction<Theme>>;
  addFavValidators?: (id: string) => void;
  deleteFavValidators?: (id: string) => void;
  addFavAddresses?: (n: FavAddress) => void;
  deleteFavAddresses?: (id: string) => void;
  updateFavAddresses?: (n: FavAddress) => void;
}

const initialState: GeneralState = {
  currency: 'USD',
  currencyRate: 1,
  theme: 'light',
  alwaysRequirePassword: false,
  hideAmount: false,
  favValidators: [],
  favAddresses: [],
};

const GeneralContext = React.createContext<GeneralState>(initialState);

const GeneralProvider: React.FC = ({ children }) => {
  const [currency, setCurrency] = usePersistedState('currency', initialState.currency);
  const [currencyRate, setCurrencyRate] = usePersistedState(
    'currencyRate',
    initialState.currencyRate,
  );
  const [theme, setTheme] = usePersistedState('theme', initialState.theme);
  const [alwaysRequirePassword, setAlwaysRequirePassword] = usePersistedState(
    'alwaysRequirePassword',
    initialState.alwaysRequirePassword,
  );
  const [hideAmount, setHideAmount] = usePersistedState('hideAmount', initialState.hideAmount);

  const [favValidators, setFavValidators] = usePersistedState('fav', initialState.favValidators);
  const [favAddresses, setFavAddresses] = usePersistedState(
    'favAddress',
    initialState.favAddresses,
  );
  const addFavValidators = React.useCallback(
    (address: string) => {
      setFavValidators(vs => [address, ...vs]);
    },
    [setFavValidators],
  );

  const deleteFavValidators = React.useCallback(
    (address: string) => {
      setFavValidators(vs => vs.filter(a => a !== address));
    },
    [setFavValidators],
  );

  const addFavAddresses = React.useCallback(
    (newAddress: FavAddress) => {
      setFavAddresses(vs => [newAddress, ...vs]);
    },
    [setFavAddresses],
  );

  const deleteFavAddresses = React.useCallback(
    (address: string) => {
      setFavAddresses(vs => vs.filter(a => a.address !== address));
    },
    [setFavAddresses],
  );

  const updateFavAddresses = React.useCallback(
    (editedAddress: {
      address: string;
      crypto: string;
      moniker: string;
      note: string;
      img: string;
      newAddress: string;
    }) => {
      setFavAddresses(vs =>
        vs.map(a =>
          a.address === editedAddress.address
            ? {
                address: editedAddress.newAddress,
                crypto: editedAddress.crypto,
                moniker: editedAddress.moniker,
                note: editedAddress.note,
                img: editedAddress.img,
              }
            : a,
        ),
      );
    },
    [setFavAddresses],
  );

  React.useEffect(() => {
    if (currency === 'USD') {
      setCurrencyRate(1);
    } else {
      fetch(
        `${process.env.NEXT_PUBLIC_ALPHAVANTAGE_API_URL}/query?function=CURRENCY_EXCHANGE_RATE&from_currency=USD&to_currency=${currency}&apikey=${process.env.NEXT_PUBLIC_ALPHAVANTAGE_API_KEY}`,
      )
        .then(r => r.json())
        .then(r =>
          setCurrencyRate(
            Number(get(r, ['Realtime Currency Exchange Rate', '5. Exchange Rate'], '')),
          ),
        );
    }
  }, [currency]);

  return (
    <GeneralContext.Provider
      value={{
        currency,
        currencyRate,
        setCurrency,
        theme,
        setTheme,
        alwaysRequirePassword,
        setAlwaysRequirePassword,
        hideAmount,
        setHideAmount,
        favValidators,
        addFavValidators,
        deleteFavValidators,
        favAddresses,
        addFavAddresses,
        deleteFavAddresses,
        updateFavAddresses,
      }}
    >
      {children}
    </GeneralContext.Provider>
  );
};

const useGeneralContext = (): GeneralState => React.useContext(GeneralContext);

export { GeneralProvider, useGeneralContext };
