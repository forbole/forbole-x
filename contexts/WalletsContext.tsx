import React from 'react'
import sendMsgToChromeExt from '../misc/sendMsgToChromeExt'

interface WalletsState {
  isFirstTimeUser: boolean
  isUnlocked: boolean
  wallets: Wallet[]
  accounts: Account[]
  password: string
  unlockWallets?: (password: string) => void
  addWallet?: (wallet: CreateWalletParams) => void
  updateWallet?: (id: string, wallet: UpdateWalletParams) => void
  deleteWallet?: (id: string) => void
  addAccount?: (account: CreateAccountParams, securityPassword: string) => void
  updateAccount?: (address: string, account: UpdateAccountParams) => void
  deleteAccount?: (address: string) => void
  viewMnemonicPhrase?: (id: string, securityPassword: string) => Promise<{ success: boolean }>
  viewMnemonicPhraseBackup?: (
    id: string,
    securityPassword: string,
    backupPassword: string
  ) => Promise<string>
}

const initialState: WalletsState = {
  isFirstTimeUser: false,
  isUnlocked: false,
  wallets: [],
  accounts: [],
  password: '',
}

const WalletsContext = React.createContext<WalletsState>(initialState)

const WalletsProvider: React.FC = ({ children }) => {
  const [wallets, setWallets] = React.useState<Wallet[]>([])
  const [accounts, setAccounts] = React.useState<Account[]>([])
  const [isFirstTimeUser, setIsFirstTimeUser] = React.useState(false)
  const [password, setPassword] = React.useState('')

  const checkIsFirstTimeUser = React.useCallback(async () => {
    const response = await sendMsgToChromeExt({
      event: 'ping',
    })
    setIsFirstTimeUser(response.isFirstTimeUser)
  }, [])

  const unlockWallets = React.useCallback(
    async (pw: string) => {
      if (!isFirstTimeUser) {
        const walletaResponse = await sendMsgToChromeExt({
          event: 'getWallets',
          data: {
            password: pw,
          },
        })
        const accountsResponse = await sendMsgToChromeExt({
          event: 'getAccounts',
          data: {
            password: pw,
          },
        })
        setWallets(walletaResponse.wallets)
        setAccounts(accountsResponse.accounts)
      }
      setPassword(pw)
    },
    [isFirstTimeUser, setPassword, setWallets, setAccounts]
  )

  const addWallet = React.useCallback(
    async (wallet: CreateWalletParams) => {
      const result = await sendMsgToChromeExt({
        event: 'addWallet',
        data: {
          wallet,
          password,
        },
      })
      setIsFirstTimeUser(false)
      setWallets((ws) => [result.wallet, ...ws])
      setAccounts((acs) => [...result.accounts, ...acs])
    },
    [password, setIsFirstTimeUser, setWallets, setAccounts]
  )

  const updateWallet = React.useCallback(
    async (id: string, wallet: UpdateWalletParams) => {
      const result = await sendMsgToChromeExt({
        event: 'updateWallet',
        data: {
          wallet,
          id,
          password,
        },
      })
      setWallets((ws) => ws.map((w) => (w.id === id ? result.wallet : w)))
    },
    [password, setWallets]
  )

  const deleteWallet = React.useCallback(
    async (id: string) => {
      await sendMsgToChromeExt({
        event: 'deleteWallet',
        data: {
          id,
          password,
        },
      })
      await Promise.all(
        accounts
          .filter((a) => a.walletId === id)
          .map((a) =>
            sendMsgToChromeExt({
              event: 'deleteAccount',
              data: {
                address: a.address,
                password,
              },
            })
          )
      )
      setWallets((ws) => {
        const newWallets = ws.filter((w) => w.id !== id)
        if (!newWallets.length) {
          setIsFirstTimeUser(true)
        }
        return newWallets
      })
      setAccounts((acs) => acs.filter((a) => a.walletId !== id))
    },
    [password, accounts, setWallets, setIsFirstTimeUser, setAccounts]
  )

  const addAccount = React.useCallback(
    async (account: CreateAccountParams, securityPassword: string) => {
      const result = await sendMsgToChromeExt({
        event: 'addAccount',
        data: {
          account,
          securityPassword,
          password,
        },
      })
      setAccounts((acs) => [result.account, ...acs])
    },
    [password, setAccounts]
  )

  const updateAccount = React.useCallback(
    async (address: string, account: UpdateAccountParams) => {
      const result = await sendMsgToChromeExt({
        event: 'updateAccount',
        data: {
          account,
          address,
          password,
        },
      })
      setAccounts((acs) => acs.map((a) => (a.address === address ? result.account : a)))
    },
    [password, setAccounts]
  )

  const deleteAccount = React.useCallback(
    async (address: string) => {
      await sendMsgToChromeExt({
        event: 'deleteAccount',
        data: {
          address,
          password,
        },
      })
      setAccounts((acs) => acs.filter((a) => a.address !== address))
    },
    [password, setAccounts]
  )

  // Can be used to verify security password
  const viewMnemonicPhrase = React.useCallback(
    async (id: string, securityPassword: string) => {
      const { mnemonic } = await sendMsgToChromeExt({
        event: 'viewMnemonicPhrase',
        data: {
          id,
          securityPassword,
          password,
        },
      })
      return mnemonic
    },
    [password]
  )

  const viewMnemonicPhraseBackup = React.useCallback(
    async (id: string, securityPassword: string, backupPassword: string) => {
      const { mnemonic } = await sendMsgToChromeExt({
        event: 'viewMnemonicPhraseBackup',
        data: {
          id,
          securityPassword,
          backupPassword,
          password,
        },
      })
      return mnemonic
    },
    [password]
  )

  React.useEffect(() => {
    checkIsFirstTimeUser()
  }, [])

  return (
    <WalletsContext.Provider
      value={{
        isFirstTimeUser,
        isUnlocked: !!wallets.length,
        wallets,
        accounts,
        password,
        unlockWallets,
        addWallet,
        updateWallet,
        deleteWallet,
        addAccount,
        updateAccount,
        deleteAccount,
        viewMnemonicPhraseBackup,
        viewMnemonicPhrase,
      }}
    >
      {children}
    </WalletsContext.Provider>
  )
}

const useWalletsContext = (): WalletsState => React.useContext(WalletsContext)

export { WalletsProvider, useWalletsContext }
