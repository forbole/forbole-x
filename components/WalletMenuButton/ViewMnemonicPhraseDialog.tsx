import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
  TextField,
  Typography,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import CloseIcon from '../../assets/images/icons/icon_cross.svg'
import BackIcon from '../../assets/images/icons/icon_back.svg'
import useStyles from './styles'
import { useWalletsContext } from '../../contexts/WalletsContext'
import useIconProps from '../../misc/useIconProps'
import PasswordInput from '../PasswordInput'
import useStateHistory from '../../misc/useStateHistory'

enum Stage {
  SecurityPassword = 'security password',
  BackupPassword = 'backup password',
  ExportMnemonic = 'export mnemonic',
}

interface ViewMnemonicPhraseDialogProps {
  walletId: string
  open: boolean
  onClose(): void
}

const ViewMnemonicPhraseDialog: React.FC<ViewMnemonicPhraseDialogProps> = ({
  walletId,
  open,
  onClose,
}) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const iconProps = useIconProps()
  const [error, setError] = React.useState('')
  const [securityPassword, setSecurityPassword] = React.useState('')
  const [backupPassword, setBackupPassword] = React.useState('')
  const [encryptionPhrase, setEncryptionPhrase] = React.useState('')
  const [stage, setStage, toPrevStage, isPrevStageAvailable] = useStateHistory<Stage>(
    Stage.SecurityPassword
  )
  const { viewMnemonicPhrase, viewMnemonicPhraseBackup } = useWalletsContext()

  const onButtonClick = React.useCallback(async () => {
    try {
      setError('')
      if (stage === Stage.SecurityPassword) {
        await viewMnemonicPhrase(walletId, securityPassword)
        setStage(Stage.BackupPassword)
      } else if (stage === Stage.BackupPassword) {
        const result = await viewMnemonicPhraseBackup(walletId, securityPassword, backupPassword)
        setEncryptionPhrase(result)
        setStage(Stage.ExportMnemonic)
      } else {
        onClose()
      }
    } catch (err) {
      setError(err.message)
    }
  }, [
    securityPassword,
    walletId,
    backupPassword,
    stage,
    setError,
    setStage,
    viewMnemonicPhraseBackup,
    viewMnemonicPhrase,
  ])

  React.useEffect(() => {
    if (open) {
      setError('')
      setSecurityPassword('')
      setBackupPassword('')
      setStage(Stage.SecurityPassword, true)
      setEncryptionPhrase('')
    }
  }, [open])

  return (
    <Dialog fullWidth open={open} onClose={onClose}>
      {isPrevStageAvailable ? (
        <IconButton className={classes.backButton} onClick={toPrevStage}>
          <BackIcon {...iconProps} />
        </IconButton>
      ) : null}
      <IconButton className={classes.closeButton} onClick={onClose}>
        <CloseIcon {...iconProps} />
      </IconButton>
      <DialogTitle>{t(`${stage} title`)}</DialogTitle>
      <DialogContent>
        {stage === Stage.SecurityPassword ? (
          <>
            <Typography className={classes.stageDescription} gutterBottom>
              {t(`${stage} description`)}
            </Typography>
            <Box mb={18}>
              <Typography gutterBottom>{t('enter security password')}</Typography>
              <PasswordInput
                value={securityPassword}
                onChange={(e) => setSecurityPassword(e.target.value)}
                placeholder={t('password')}
                error={!!error}
                helperText={error}
              />
            </Box>
          </>
        ) : null}
        {stage === Stage.BackupPassword ? (
          <Box mb={6}>
            <Typography gutterBottom>{t('encryption password')}</Typography>
            <PasswordInput
              value={backupPassword}
              onChange={(e) => setBackupPassword(e.target.value)}
              placeholder={t('password')}
              error={!!error}
              helperText={error}
            />
          </Box>
        ) : null}
        {stage === Stage.ExportMnemonic ? (
          <Box mb={4}>
            <TextField
              variant="filled"
              InputProps={{ disableUnderline: true }}
              fullWidth
              value={encryptionPhrase}
              multiline
              rows={5}
              onFocus={(e: any) => {
                e.target.select()
                navigator.clipboard.writeText(encryptionPhrase)
              }}
            />
          </Box>
        ) : null}
      </DialogContent>
      <DialogActions>
        <Button
          className={classes.dialogButton}
          variant="contained"
          color="primary"
          onClick={onButtonClick}
          disabled={stage === Stage.BackupPassword && backupPassword.length < 6}
        >
          {t(stage === Stage.ExportMnemonic ? 'confirm' : 'next')}
        </Button>
      </DialogActions>
    </Dialog>
  )
}

export default ViewMnemonicPhraseDialog
