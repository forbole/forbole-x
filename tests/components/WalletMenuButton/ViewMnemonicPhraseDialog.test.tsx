import React from 'react'
import renderer from 'react-test-renderer'
import last from 'lodash/last'
import { useWalletsContext } from '../../../contexts/WalletsContext'
import ViewMnemonicPhraseDialog from '../../../components/WalletMenuButton/ViewMnemonicPhraseDialog'

const onClose = jest.fn()
const viewMnemonicPhraseBackup = jest.fn().mockResolvedValue('encrypted mnemonic')
const viewMnemonicPhrase = jest.fn()

const selectInput = jest.fn()
const copyText = jest.fn()
Object.defineProperty(navigator, 'clipboard', {
  value: {
    writeText: copyText,
  },
})

jest.mock('@material-ui/core/Dialog', () => (props) => <div id="dialog" {...props} />)
jest.mock('../../../contexts/WalletsContext', () => ({
  useWalletsContext: jest.fn(),
}))
jest.mock('../../../components/PasswordInput', () => 'input')
;(useWalletsContext as jest.Mock).mockReturnValue({ viewMnemonicPhraseBackup, viewMnemonicPhrase })

describe('component: ViewMnemonicPhraseDialog', () => {
  it('renders opened state correctly', () => {
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open onClose={onClose} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders closed state correctly', () => {
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open={false} onClose={onClose} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onClose on close button click', () => {
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open onClose={onClose} />
    )
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick({ currentTarget: 'anchor' })
    })
    expect(onClose).toBeCalled()
  })
  it('calls viewMnemonicPhrase on next button click, then goes back on back button click', async () => {
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open onClose={onClose} />
    )
    renderer.act(() => undefined)
    renderer.act(() => {
      component.root.findByType('input').props.onChange({ target: { value: 'password' } })
    })
    await renderer.act(async () => {
      await last(component.root.findAllByType('button')).props.onClick()
    })
    expect(viewMnemonicPhrase).toBeCalledWith('123', 'password')
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls viewMnemonicPhrase on next button click, then calls viewMnemonicPhraseBackup on next button click, then calls onClose on next button click', async () => {
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open onClose={onClose} />
    )
    renderer.act(() => undefined)
    renderer.act(() => {
      component.root.findByType('input').props.onChange({ target: { value: 'password' } })
    })
    await renderer.act(async () => {
      await last(component.root.findAllByType('button')).props.onClick()
    })
    expect(viewMnemonicPhrase).toBeCalledWith('123', 'password')
    renderer.act(() => {
      component.root
        .findByType('input')
        .props.onChange({ target: { value: 'encryption password' } })
    })
    await renderer.act(async () => {
      await last(component.root.findAllByType('button')).props.onClick()
    })
    expect(viewMnemonicPhraseBackup).toBeCalledWith('123', 'password', 'encryption password')
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    renderer.act(() => {
      component.root.findByType('textarea').props.onFocus({ target: { select: selectInput } })
    })
    expect(selectInput).toBeCalled()
    expect(copyText).toBeCalledWith('encrypted mnemonic')
    await renderer.act(async () => {
      await last(component.root.findAllByType('button')).props.onClick()
    })
    expect(onClose).toBeCalled()
  })
  it('handles error on next button click', async () => {
    viewMnemonicPhrase.mockRejectedValueOnce(new Error('error'))
    const component = renderer.create(
      <ViewMnemonicPhraseDialog walletId="123" open onClose={onClose} />
    )
    renderer.act(() => undefined)
    renderer.act(() => {
      component.root.findByType('input').props.onChange({ target: { value: 'password' } })
    })
    await renderer.act(async () => {
      await component.root.findAllByType('button')[1].props.onClick()
    })
    expect(viewMnemonicPhrase).toBeCalledWith('123', 'password')
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
