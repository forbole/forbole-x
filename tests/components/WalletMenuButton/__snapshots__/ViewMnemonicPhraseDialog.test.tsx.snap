// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: ViewMnemonicPhraseDialog calls viewMnemonicPhrase on next button click, then calls viewMnemonicPhraseBackup on next button click, then calls onClose on next button click 1`] = `
<div
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-backButton-5"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_back"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:export mnemonic title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root"
  >
    <div
      className="MuiBox-root MuiBox-root-18"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            rows={5}
            value="encrypted mnemonic"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-dialogButton-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:confirm
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`component: ViewMnemonicPhraseDialog calls viewMnemonicPhrase on next button click, then goes back on back button click 1`] = `
<div
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:security password title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root"
  >
    <p
      className="MuiTypography-root HookGlobalStyles-stageDescription-7 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-15"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        common:enter security password
      </p>
      <input
        error={false}
        helperText=""
        onChange={[Function]}
        placeholder="common:password"
        value="password"
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-dialogButton-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:next
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`component: ViewMnemonicPhraseDialog handles error on next button click 1`] = `
<div
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:security password title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root"
  >
    <p
      className="MuiTypography-root HookGlobalStyles-stageDescription-7 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-19"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        common:enter security password
      </p>
      <input
        error={true}
        helperText="error"
        onChange={[Function]}
        placeholder="common:password"
        value="password"
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-dialogButton-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:next
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`component: ViewMnemonicPhraseDialog renders closed state correctly 1`] = `
<div
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={false}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:security password title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root"
  >
    <p
      className="MuiTypography-root HookGlobalStyles-stageDescription-7 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        common:enter security password
      </p>
      <input
        error={false}
        helperText=""
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-dialogButton-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:next
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`component: ViewMnemonicPhraseDialog renders opened state correctly 1`] = `
<div
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:security password title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root"
  >
    <p
      className="MuiTypography-root HookGlobalStyles-stageDescription-7 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        common:enter security password
      </p>
      <input
        error={false}
        helperText=""
        onChange={[Function]}
        placeholder="common:password"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-dialogButton-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:next
      </span>
    </button>
  </div>
</div>
`;
